### DISCUSSION TITLE
Unable to run EksPodOperator since updating mwaa from 2.4.3 to 2.7.1

### DISCUSSION THREAD

**Discussion by 0x26res** [apache/airflow#37178] [Q&A]
While upgrading aws MWAA from 2.4.3 to 2.7.1, my `EksPodOperator` stopped working with this error:

```
HTTP response headers: HTTPHeaderDict({'Audit-Id': '7d78edc7-c595-4baa-8a54-', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'X-Kubernetes-Pf-Flowschema-Uid': '5c0c1226-e427-42a3-ab4c-', 'X-Kubernetes-Pf-Prioritylevel-Uid': '5c640447-3651-46ec-8ad2-', 'Date': 'Mon, 05 Feb 2024 00:15:24 GMT', 'Content-Length': '270'})
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods is forbidden: User \"system:anonymous\" cannot list resource \"pods\" in API group \"\" in the namespace \"shared\"","reason":"Forbidden","details":{"kind":"pods"},"code":403}
```

I haven't changed any of the permissions, so I believe some changes in the EksPodOperator, KubernetesPodOperator or any hook they use is causing the issue. 

I've been following these [instructions](https://docs.aws.amazon.com/mwaa/latest/userguide/mwaa-eks-example.html) though they are not completely relevant since they use the KubernetesPodOperator instead of the EksPodOperator

This upgrade took me:
- from apache-airflow-providers-cncf-kubernetes==4.4.0 to apache-airflow-providers-cncf-kubernetes==7.6.0.
- from apache-airflow-providers-amazon==6.0.0 to apache-airflow-providers-amazon==8.7.1

**Comment by kl4072** [comment]
were you able to resolve this? 
I have similar issue with AKS and running 2.8.1 the service account has access to the namespace its deployed in, but it is trying to list default namespace which it wouldnt have access on in my case.

**Comment by meshok0** [comment]
We had a similar issue(message":"pods is forbidden: User \"system:anonymous\" cannot list resource \"pods\" in tasks logs) on MWAA + EKS on Fargate. 
AWS support suggested to change the verbosity of the worker log group of the MWAA environment to WARNING or ERROR. Unbelievable, but this fixed the issue.

**Reply by 0x26res** [reply]
Hi @meshok0, just to confirm, did you change the verbosity from the default (INFO) to WARNING and it worked?

**Comment by meshok0** [comment]
Yes

**Reply by 0x26res** [reply]
Ok, I gave it a try, it works. Thanks for pointing it out.

There's a big caveat: you can't submit deferrable task. I'll create an issue.

**Reply by 0x26res** [reply]
I've created an issue for the new problem I have encountered: https://github.com/apache/airflow/issues/38280

**Comment by andrewhharmon** [comment]
Im facing this problem as well but I don't understand the fix. Why would changing the verbosity of the worker logs fix this.

**Reply by 0x26res** [reply]
I don't have a good explanation either. It could be that the debug log tries to query Eks for more information and that's when the failure happens.

But I can confirm that changing the log level fixes the issue.

**Comment by evredinka** [comment]
I faced the same issue upgrading from version 2.0.2→2.7.2 and 2.0.2→2.8.1 and the issue still remained even after logging level adjustment.
The root cause in my case was outdated `kube_config.yaml`:
<img width="1059" alt="Screenshot 2024-06-24 at 21 29 22" src="https://github.com/apache/airflow/assets/12608352/c29d6637-bfec-4d82-bccc-f8ca9d062b43">

**Comment by andrewhharmon** [comment]
mine ended up being network related. Even though the EKS API Server is public facing, it was still taking a private route. So I had to open access in the security group. https://stackoverflow.com/questions/78638015/aws-mwaa-times-out-trying-to-connect-to-eks-api-server/78638114#78638114
